APCAppCore/APCAppCore/Library/FeedParser/APCNewsFeedManager.m

```diff
@@ -0,0 +1,159 @@
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+
+#import "APCNewsFeedManager.h"
+#import "APCFeedParser.h"
+#import "APCLog.h"
+
+NSString * const kAPCNewsFeedUpdateNotification = @"APCNewsFeedUpdateNotification";
+
+static NSString * const kAPCReadPostsKey = @"ReadPostsKey";
+static NSString * const kAPCBlogUrlKey   = @"BlogUrlKey";
+
+@interface APCNewsFeedManager()
+
+@property (nonatomic, strong) APCFeedParser *feedParser;
+
+@property (nonatomic, strong) NSArray *readPosts;
+
+@end
+
+@implementation APCNewsFeedManager
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        
+        NSString *urlString = [self blogUrlFromJSONFile:@"StudyOverview"];
+        _feedParser = [[APCFeedParser alloc] initWithFeedURL:[NSURL URLWithString:urlString]];
+        
+        NSString *savedBlogUrl = [[NSUserDefaults standardUserDefaults] stringForKey:kAPCBlogUrlKey];
+        
+        //Clear read links if blog URL has changed.
+        if (![savedBlogUrl isEqualToString:urlString]) {
+            [[NSUserDefaults standardUserDefaults] removeObjectForKey:kAPCReadPostsKey];
+            
+            [[NSUserDefaults standardUserDefaults] setObject:urlString forKey:kAPCBlogUrlKey];
+            [[NSUserDefaults standardUserDefaults] synchronize];
+        }
+        
+    }
+    return self;
+}
+
+/**********************************/
+#pragma mark - Public methods
+/**********************************/
+
+- (void)fetchFeedWithCompletion:(APCNewsFeedManagerCompletionBlock)completion
+{
+    __weak typeof(self) weakSelf = self;
+    
+    [self.feedParser fetchFeedWithCompletion:^(NSArray *results, NSError *error) {
+        
+        weakSelf.feedPosts = results;
```

Comment:
As @jwe-apple suggested, to prevent `self` instance being released during the execution of current block, convert `weakSelf` to `strongSelf`to retain `self`.

```
__strong __typeof(self) strongSelf = weakSelf;
strongSelf.feedPosts = results;
```